File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/ContextEvaluator.java
package com.github.iamrenny.ruleflow.evaluators;

import com.github.iamrenny.ruleflow.errors.PropertyNotFoundException;
import com.github.iamrenny.ruleflow.visitors.Visitor;
import org.antlr.v4.runtime.ParserRuleContext;

public interface ContextEvaluator<T extends ParserRuleContext> {
    Object evaluate(T ctx, Visitor visitor) throws PropertyNotFoundException;
}

File: rules-interpreter/src/main/kotlin/com/rappi/fraud/rules/parser/evaluators/ContextEvaluator.kt
package com.rappi.fraud.rules.parser.evaluators

import com.rappi.fraud.rules.parser.visitors.Visitor
import org.antlr.v4.runtime.ParserRuleContext

interface ContextEvaluator<T: ParserRuleContext> {

    fun evaluate(ctx: T, visitor: Visitor): Any
}